gtsfm/multi_view_optimizer.py

```diff
@@ -80,8 +83,15 @@ def create_computation_graph(
 
         ba_result_graph = self.ba_optimizer.create_computation_graph(ba_input_graph)
 
+        if gt_poses_graph is not None:
+            self.metrics_graph = dask.delayed(metrics.save_averaging_metrics)(
```

Comment:
This is a clean design, but I dont like the fact the function `create_computation_graph` changes the attributes of the class. As we sticking with a functional paradigm, I'd like no side-effects